CONVENTION.md

```diff
@@ -0,0 +1,453 @@
+
+<br>
+
+#### Structure I - View
+
+This involves the `def __init__()` in `view.py`, the remaining methods being signal and event handlers.
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+    	self.setWindowTitle(self.title)
+        self.setWindowIcon(QtGui.QIcon(...))
+        self.setAttribute(QtCore.Qt.WA_StyledBackground)
+
+    	pages = {
+    		"home": QtWidgets.QWidget(),
+    	}
+
+    	panels = {
+    		"header": QtWidgets.QWidget(),
+    		"body": QtWidgets.QWidget(),
+    		"footer": QtWidgets.QWidget(),
+    		"sidebar": QtWidgets.QWidget(),
+    	}
+
+    	widgets = {
+    		"pages": QtWidgets.QStackedWidget(),
+    		"logo": QtWidgets.QWidget(),
+    		"okBtn": QtWidgets.QPushButton("Ok"),
+    		"cancelBtn": QtWidgets.QPushButton("Cancel"),
+    		"resetBtn": QtWidgets.QPushButton("Reset"),
+    	}
+
+    	icons = {
+    		"cancelBtn": resources.icon("cancelBtn"),
+    	}
+
+```
+
+**Notes**
+
+- **S1** Widgets MUST are separated into `pages`, `panels` and `widgets`
+	- Pages are the full screen of an application, except statusbar, docks, toolbar and menu. Most apps only have one page, but some - like Pyblish Lite and QML - have more
+	- Panels are subdivisions of a Page
+	- Widgets are self explanatory
+- **S2** Widgets MUST be declared at the top
+- **S3** Widgets MUST have a mixedCase name, used in CSS
+- **S4** `Window` MUST have `title` declared as class attribute, used to dynamically change window title at run-time, e.g. to reflect a selection or state
+- **S5** Like widgets, icons SHOULD be declared up-front
+- **S6** Resource logic and I/O SHOULD be handled separately, for testing
+- **S7** Object names SHOULD be mixedCase, to conform with JavaScript/CSS where they are used
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+
+		...
+
+        for name, widget in itertools.chain(panels.items(),
+                                            widgets.items(),
+                                            pages.items()):
+            # Expose to CSS
+            widget.setObjectName(name)
+
+        	# Support for CSS
+            widget.setAttribute(QtCore.Qt.WA_StyledBackground)
+
+        self.setCentralWidget(widgets["pages"])
+        widgets["pages"].addWidget(pages["home"])
+
+```
+
+**Notes**
+
+- **S10** Every widget MUST be given a unique name
+- **S11** Every widget MUST be stylable with CSS
+- **S12** Central widget MUST be a `QStackedWidget` to facilitate multiple pages 
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+
+    	...
+
+    	layout = QtWidgets.QHBoxLayout(panels["header"])
```

Comment:
Are we sure it reads better to use `panels["header"]` than just `header` as a variable? It's not a standard practice I've seen before.

Reply:
> Are we sure it reads better to use panels["header"] than just header as a variable?

Yeah, let's stick with that. It's something that has worked well in many differently sized UIs, and remains consistent and safe. The added bonus is that it also gives the reader a clear indication of what the name in CSS will be, and that when it's accessed from elsewhere - which you can expect it can - it'll be `self._panels["header"]` as opposed to.. `self._header?` What is that, a line of text? A widget, or what?

Reply:
It's both fine for me. Just wondered whether others didn't feel this would turn into a clutter, including yourself. If not, fine with both to adapt as needed.